containers/unit_tests/TestDualView.hpp

```diff
@@ -87,17 +87,6 @@ struct test_dualview_copy_construction_and_assignment {
 
     ASSERT_EQ(a.view_host(), c.view_host());
     ASSERT_EQ(a.view_device(), c.view_device());
-
-    // We can't test shallow equality of modified_flags because it's protected.
-    // So we test it indirectly through sync state behavior.
-    if (!std::decay_t<SrcViewType>::impl_dualview_stores_single_view) {
-      a.clear_sync_state();
-      a.modify_host();
-      ASSERT_TRUE(a.need_sync_device());
-      ASSERT_TRUE(b.need_sync_device());
-      ASSERT_TRUE(c.need_sync_device());
-      a.clear_sync_state();
-    }
```

Comment:
What happened here?

Reply:
I reverted the adding `DualView::impl_dualview_stores_single_view`

Reply:
This test was added in https://github.com/kokkos/kokkos/pull/6474 so I would keep it the test (replacing `impl_dualview_stores_single_view` with `impl_dualview_is_single_device`) or also revert #6474 for consistency.